datafusion/sql/src/statement.rs

```diff
@@ -40,14 +40,39 @@ use datafusion_expr::{
 };
 use sqlparser::ast;
 use sqlparser::ast::{
-    Assignment, Expr as SQLExpr, Expr, Ident, ObjectName, ObjectType, Query, SetExpr,
-    ShowCreateObject, ShowStatementFilter, Statement, TableFactor, TableWithJoins,
-    UnaryOperator, Value,
+    Assignment, Expr as SQLExpr, Expr, Ident, ObjectName, ObjectType, Query, SchemaName,
+    SetExpr, ShowCreateObject, ShowStatementFilter, Statement, TableFactor,
+    TableWithJoins, UnaryOperator, Value,
 };
 use sqlparser::parser::ParserError::ParserError;
 use std::collections::{BTreeMap, HashMap, HashSet};
 use std::sync::Arc;
 
+fn ident_to_string(ident: &Ident) -> String {
+    ident.value.to_owned()
+}
+
+fn object_name_to_string(object_name: &ObjectName) -> String {
+    object_name
+        .0
+        .iter()
+        .map(ident_to_string)
+        .collect::<Vec<String>>()
+        .join(".")
+}
+
+fn get_schema_name(schema_name: &SchemaName) -> String {
+    match schema_name {
+        SchemaName::Simple(schema_name) => object_name_to_string(schema_name),
+        SchemaName::UnnamedAuthorization(auth) => ident_to_string(auth),
+        SchemaName::NamedAuthorization(schema_name, auth) => format!(
+            "{}.{}",
+            object_name_to_string(schema_name),
```

Comment:
I wonder if this name needs to be normalized too

like `CREATE SCHEMA Foo` should create the schema `foo` (not `Foo`)

Reply:
Hey, thank you so much for the quick and positive response!
Honestly, I have no idea if `Foo` should become `foo`, but I'm happy to update the code to call `utils::normalize_ident()` on `ident.value` if necessary!

Reply:
Yes I think it should be lowercase, here is a demonstration of how that works in postgres:

```sql
postgres=# drop schema foo;
DROP SCHEMA
postgres=# create schema foo;
CREATE SCHEMA
postgres=# drop schema foo;
DROP SCHEMA
postgres=# create schema FoO;
CREATE SCHEMA
postgres=# create schema foo;
ERROR:  schema "foo" already exists
```

Reply:
Thanks for the input! The code in the PR now calls `normalize_ident` and includes an integration test asserting this behavior.